### ISSUE TITLE
Converting to png gives black image but thumbnail is fine?

### ISSUE THREAD

**Issue by fenchai23** [DavBfr/dart_pdf#392]
Simple hello world example
```
final pw.Document pdf = pw.Document();

    pdf.addPage(pw.Page(
        pageFormat: PdfPageFormat.a4,
        build: (pw.Context context) {
          return pw.Center(
            child: pw.Text("Hello World"),
          ); // Center
        }));

final output = await getTemporaryDirectory();

    final file = File("${output.path}/cotizacion.png");

    await for (var page in Printing.raster(pdf.save(), dpi: 72)) {
      final image = await page.toPng();

      await file.writeAsBytes(image);

      if (openFile) {
        OpenFile.open(file.path);
      }
    }
```

gives black image?
![image](https://user-images.githubusercontent.com/52770796/90072606-5cad1c80-dcbd-11ea-9414-c802815f1878.png)

but the thumbnail seems fine?
![image](https://user-images.githubusercontent.com/52770796/90072663-73ec0a00-dcbd-11ea-91b8-a048c7efb416.png)

is there something I'm doing wrong here? thanks

**pdf works** converting to png gives black screen.
**pdf:** ^1.6.0
**printing:** ^3.6.0

**Comment by DavBfr** [comment]
Maybe you need some permissions on this file to open it with another application.

**Comment by fenchai23** [comment]
> Maybe you need some permissions on this file to open it with another application.

mmm it already has read and write, I don't think that's it because it's just using the cache locations.

I opened the file on Android Studio and it seems the images I inserted on the pdf is shown, also the widgets with background colors are also shown.

I think it's missing background color? I can see transparency showing in the image.

How do I add white background color to the pdf? (well soon to be converted into png)

I'm sure this is the problem.

**Comment by fenchai23** [comment]
yeap, fixed by doing this https://github.com/DavBfr/dart_pdf/issues/168#issuecomment-558167447
seems to have no margin settings now even thought these are my margin settings:

```
pageTheme: pw.PageTheme(
          buildBackground: (pw.Context context) => pw.Container(
            color: PdfColors.white,
          ),
          pageFormat: PdfPageFormat.letter.copyWith(
            marginBottom: 1.5 * PdfPageFormat.cm,
            marginTop: 1.5 * PdfPageFormat.cm,
            marginLeft: 1 * PdfPageFormat.cm,
            marginRight: 1 * PdfPageFormat.cm,
          ),
        ),
```

**Comment by DavBfr** [comment]
Yes, the generated PNG file is transparent where there is nothing to print and the OpenFile plugin has a black background.

**Comment by fenchai23** [comment]
> Yes, the generated PNG file is transparent where there is nothing to print and the OpenFile plugin has a black background.

anyway to fix the low res png file? or should I open another issue?

**Comment by DavBfr** [comment]
The DPI setting will allow you to have a higher resolution. With 75 DPI it will be blurry, try a higher value.

**Comment by fenchai23** [comment]
> The DPI setting will allow you to have a higher resolution. With 75 DPI it will be blurry, try a higher value.

thanks, will increase dpi.

also thanks for creating pdf and printing packages :) one more coffee to the array!

**Comment by hongquang198** [comment]
> yeap, fixed by doing this [#168 (comment)](https://github.com/DavBfr/dart_pdf/issues/168#issuecomment-558167447) seems to have no margin settings now even thought these are my margin settings: ok it seems there are margins, BUT the margins are **transpararent** now. Anyway to theme the margins? also fixed by using **pw.FullPage** on buildBackground. Now the image seems to be a bit blurry, I have set the DPI to be 75 using letter as the page size. I will keep testing it.
> 
> ```
> pageTheme: pw.PageTheme(
>           buildBackground: (pw.Context context) => pw.Container(
>             color: PdfColors.white,
>           ),
>           pageFormat: PdfPageFormat.letter.copyWith(
>             marginBottom: 1.5 * PdfPageFormat.cm,
>             marginTop: 1.5 * PdfPageFormat.cm,
>             marginLeft: 1 * PdfPageFormat.cm,
>             marginRight: 1 * PdfPageFormat.cm,
>           ),
>         ),
> ```

This fixes my problem. My printer always prints blank pages. Now with a white background color everything is printed out correctly.
Thanks so much !
